### PR TITLE
Reorg install and plugin docs

### LINKED ISSUE #1396: Add `contrib` section to repo with community-contributed examples/etc.
We want to encourage content to be contributed from the community (e.g. setup instructions for specific S3-compatible object stores, alternate YAML configs, etc.) but we want to be clear about what the core team officially supports vs. doesn't.

Let's add a directory somewhere in the repo where these things can easily be added.  As part of this, we should review what's currently in our repo and move any relevant content in there. For example, we have `docs/ibm-config.md` which was added by a contributor, but is not supported by the core team. We should move it.

### PR DESCRIPTION
**Pull request by carlisia**
Closes #1396 

Signed-off-by: Carlisia <carlisia@vmware.com>

### CODE REVIEW

**Comment by carlisia** [comment]
Note: I only made changes to the `master` location for now.

**Comment by carlisia** [comment]
Another note: I spent a ton of time trying to figure out why these links are not working, they are adding the name of the page they are on to the URL path:

image

😠 

The links should be right? (Adding `.md` doesn't fix it):
https://github.com/heptio/velero/pull/1916/files#diff-bf1c70028a3cddfecef3a254f47c84a7R17

The same is happening with the links to the ibm and oracle on the community-plugin page.

**Comment by carlisia** [comment]
Instead of adding the oracle and ibm docs to a `contrib` folder in the repo, I added here, and then rearranged the docs to spruce things up + split things into velero supported vs. community plugin.

If this works for everyone, when we move the repo we'll only have to update the links for the velero supported plugins to point to the URL in the new plugin repos (as well as remove the provider docs).

File: site/docs/master/install-overview.md [hunk 1]
```diff
@@ -26,12 +26,7 @@ velero install \
 
 When using node-based IAM policies, `--secret-file` is not required, but `--no-secret` is required for confirmation.
 
-For provider-specific instructions, see:
-
-* [Run Velero on AWS][0]
-* [Run Velero on GCP][1]
-* [Run Velero on Azure][2]
-* [Use IBM Cloud Object Store as Velero's storage destination][4]
+For provider-specific instructions, see the navigation on the left under `Plugins`.
```
Comment (skriss) [inline site/docs/master/install-overview.md]: Personally, I think it'd be easier for the user to navigate the install docs if we added *more* content to this Install Overview page about the compatible platforms and directly linked to instructions for each from here, rather than requiring them to jump into a whole different section on plugins to find links to install instructions for their platform. I think this page itself could use some reorg, maybe something like:

- Background/Prerequisites
    - Compatible object storage system, optional compatible block storage system(s), restic, ...
    - `velero install` or Helm chart
- Provider-specific instructions
    - Velero team-supported
    - Community
- `velero install` reference (details on all the different flags)

Thoughts?

**Comment by carlisia** [comment]
@jonasrosland are you aware of what this is: https://github.com/heptio/velero/pull/1916#issuecomment-535782904

**Comment by skriss** [comment]
I'm not exactly sure what's going on with the links, but for Cormac's blog posts we had to do a few `../<page>`-style links -- because individual pages' URLs look like e.g. `https://velero.io/docs/v1.1.0/upgrade-to-1.1/` (note trailing slash) which probably looks like a directory when adding on a relative link.

**Comment by carlisia** [comment]
K, I'll check on that. The links/pages look exactly like the others. But I'll check. 🤷‍♀

File: site/docs/master/install-overview.md (hunk 1 above)
Comment (carlisia) [reply site/docs/master/install-overview.md]: I made a proposed outline here based on feedback: https://hackmd.io/UpamNjb3SR2pSaJfeeHfzA

My thinking: let's nail down on this PR:
- the reorg as far as an outline
- move the existing content to fit this outline
- any writing/rewriting of content we can do in a new PR.

Comment (skriss) [reply site/docs/master/install-overview.md]: sounds like a good plan to me! taking a look at the outline shortly

Comment (skriss) [reply site/docs/master/install-overview.md]: The outline looks about right to me, I added a couple comments in the HackMD.

Comment (carlisia) [reply site/docs/master/install-overview.md]: Read your comments, make sense to me.

**Comment by carlisia** [comment]
@skriss / @nrb, I reference one of the new docs in this PR on the blog post, PTAL when you have a chance so we can merge it before the post.

**Comment by skriss** [comment]
You could also just point the blog link to the existing `support-matrix.md` page, and then change the blog link as part of this PR, if you don't want to block the blog on merge of this PR.

I'll review post-community mtg.

File: README.md [hunk 2]
```diff
@@ -15,16 +15,7 @@ Velero consists of:
 * A server that runs on your cluster
 * A command-line client that runs locally
 
-You can run Velero in clusters on a cloud provider or on-premises. For detailed information, see [Compatible Storage Providers][99].
-
-## Installation
```
Comment (skriss) [inline README.md]: What's the thinking for removing this section?

File: README.md [hunk 3]
```diff
@@ -51,10 +42,11 @@ See [the list of releases][6] to find out about feature changes.
 [11]: https://kubernetes.io/docs/tasks/tools/install-kubectl/#tabset-1
 [12]: https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/README.md
 [14]: https://github.com/kubernetes/kubernetes
+[17]: /site/docs/master/restic.md
```
Comment (skriss) [inline README.md]: Probably want to link to the velero.io website here

File: site/_data/master-toc.yml [hunk 4]
```diff
@@ -4,27 +4,33 @@ toc:
...
-      - page: Run on Azure
-        url: /azure-config
-      - page: Run on GCP
-        url: /gcp-config
-      - page: Restic setup
+      - page: Requirements
+        url: /install-requirements
+      - page: Installing
+        url: /installing
+      - page: Supported providers
```
Comment (skriss) [inline site/_data/master-toc.yml]: Do you think "supported platforms" is more clear here? I could go with either.

File: site/_data/master-toc.yml [hunk 5]
```diff
@@ -4,27 +4,33 @@ toc:
...
         url: /locations
-      - page: Supported platforms
-        url: /support-matrix
   - title: Install
     subfolderitems:
       - page: Overview
         url: /install-overview
       - page: Upgrade to 1.1
         url: /upgrade-to-1.1
-      - page: Quick start with in-cluster MinIO
```
Comment (skriss) [inline site/_data/master-toc.yml]: The idea with the quick start was to have a place for new users to quickly get up and running for evaluation/experimentation - by removing it from the TOC, that guide becomes buried. Personally I'd opt to keep it around in the TOC - what do y'all think?

File: site/_data/master-toc.yml [hunk 6]
```diff
@@ -4,27 +4,33 @@ toc:
...
+        url: /installing
+      - page: Supported providers
+        url: /supported-providers
+      - page: Restic integration
         url: /restic
+      - page: Examples
+        url: /examples
+      - page: Uninstalling
+        url: /uninstalling
+  - title: Plugins
```
Comment (skriss) [inline site/_data/master-toc.yml]: I'd move "Plugins" below "Use" in the TOC.

**Comment by carlisia** [comment]
> You could also just point the blog link to the existing support-matrix.md page, and then change the blog link as part of this PR, if you don't want to block the blog on merge of this PR.

Duh, of course! I'll do that. So let's not worry about this PR for the blog post. But still we have the tag issue.

File: site/docs/master/installing.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
+# Installing Velero
```
Comment (skriss) [inline site/docs/master/installing.md]: I'm finding this page to be confusing as-is, as it's kind of a random collection of information related to install that doesn't actually have any specific installation instructions. I think I'd prefer to either:
- move this content into the "Install Overview" page, as background (if we choose this, I bet some of the content could be removed to shorten it up as it's probably redundant with other spots)
- call this "Install Reference" or something other than just "Installing" (both the page name and in the TOC)
- reorganize this page so that it functions better as a main "installing" page

what do others think?

File: site/docs/master/supported-providers.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
+# Providers
+
+Velero can store its backups in any S3 or S3 compatible object storage.
```
Comment (skriss) [inline site/docs/master/supported-providers.md]: This doesn't make sense as a headline here since object stores other than S3 are supported, and also this page covers volume snapshot operations as well. I'd go with something more similar to what we already have on the existing page, e.g. "Velero supports a variety of storage providers for different backup and snapshot operations. Velero has a plugin system which allows anyone to add compatibility for additional backup and volume storage platforms without modifying the Velero codebase."

File: site/docs/master/supported-providers.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+
+## S3-Compatible object store providers
```
Comment (skriss) [inline site/docs/master/supported-providers.md]: I would move this below the "Velero supported providers" section

File: site/docs/master/supported-providers.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+
+Velero uses [Amazon's Go SDK][0] to connect to the AWS S3 API. Some third-party storage providers also support the S3 API, and users have reported the following providers work with Velero:
```
Comment (skriss) [inline site/docs/master/supported-providers.md]: We might want to reword this to say "Velero's AWS Object Store plugin uses Amazon's Go SDK..."

File: site/docs/master/installing.md (hunk 7 above)
Comment (skriss) [reply site/docs/master/installing.md]: To add more context here - since this page is called "Installing" in the TOC, I'd expect it to be the main page that has step-by-step instructions or well-organized links to other pages, but it's not really that at the moment, hence the ideas for different organization.

File: README.md (hunk 2 above)
Comment (carlisia) [reply README.md]: To avoid having install instructions scarred in multiple places. The content under "More information" should be all that is necessary, and should be very high level pointing to one entry point, imo. But maybe we can make this better.

File: site/_data/master-toc.yml (hunk 4 above)
Comment (carlisia) [reply site/_data/master-toc.yml]: What is the difference between platform and provider?

File: site/_data/master-toc.yml (hunk 5 above)
Comment (carlisia) [reply site/_data/master-toc.yml]: I guess I resist this because that documentation is not for production, and it would be the only thing not production worthy in this entire section. Could we add it as a `Quick start` submenu to either the `Use` or `Contributing` section?

File: site/docs/master/supported-providers.md (hunk 8 above)
Comment (carlisia) [reply site/docs/master/supported-providers.md]: Thanks for the explanation.

File: site/docs/master/installing.md (hunk 7 above)
Comment (carlisia) [reply site/docs/master/installing.md]: Ok. I see your point and agree. I'm thinking about it.

File: site/_data/master-toc.yml (hunk 4 above)
Comment (skriss) [reply site/_data/master-toc.yml]: I'd consider them roughly synonyms, but I know you disliked "provider" so was looking for alternatives

File: README.md (hunk 2 above)
Comment (skriss) [reply README.md]: OK, I guess since we now have `velero install` rather than static YAML this disclaimer isn't so important anymore. WFM.

File: site/_data/master-toc.yml (hunk 5 above)
Comment (skriss) [reply site/_data/master-toc.yml]: Personally I'd rather have it higher up since the point of it is to be a good starting point for new users. Let's see what @nrb thinks. Also FYI Istio has something similar in their Install section (I think their docs are pretty reasonably organized): https://istio.io/docs/setup/install/

File: site/_data/master-toc.yml (hunk 4 above)
Comment (carlisia) [reply site/_data/master-toc.yml]: I think minimizing the numbers of terms used for the same thing makes the documentation easier to follow.

As for `provider`, I'm starting to think we need to make more obvious the provider x plugins relationship (a provider has many plugins, a plugin has many providers). WIP.

File: site/_data/master-toc.yml (hunk 5 above)
Comment (carlisia) [reply site/_data/master-toc.yml]: Ok, I like `evaluation install`, that fits.

We can also move the Helm installation instructions up to a subitem if we can add more content to it. Not wanting to do that now tho.

File: site/_data/master-toc.yml (hunk 4 above)
Comment (skriss) [reply site/_data/master-toc.yml]: :+1: OK I'm happy with the existing title.

Comment (nrb) [reply site/_data/master-toc.yml]: I'm still not quite sure where the provider confusion is coming from ...I use it as short hand for "cloud provider."

A cloud provider can have many services. A couple of those services are used by Velero - object storage and volume snapshotting. Thus, a provider could have 1 plugin for each of those 2 services, for a total of 2. These two plugins can both live in the same binary, and the correct one will be invoked by our plugin machinery. Many others could also be in that binary (as seen in Velero's included plugins).

Things such as Portworx and Minio offer only one of those two services - volumes for Portworx, object storage for Minio. Thus they'd each only have 1 plugin, which may or may not be the only plugin within the binary. 

In Velero's API, we don't provide a field for the plugin's type, using the struct type to differentiate. This does mean that solely the names aren't enough to know what a plugin does.

File: site/_data/master-toc.yml (hunk 5 above)
Comment (nrb) [reply site/_data/master-toc.yml]: In my own reading, a "Quick start" isn't meant to be production ready. But I like the "Evaluation install" title and some wording inside the page itself.

File: README.md [hunk 11]
```diff
@@ -51,10 +42,11 @@ See [the list of releases][6] to find out about feature changes.
 [11]: https://kubernetes.io/docs/tasks/tools/install-kubectl/#tabset-1
 [12]: https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/README.md
 [14]: https://github.com/kubernetes/kubernetes
+[17]: https://velero.io/docs/master/restic.md
```
Comment (skriss) [inline README.md]: Doesn't look like this is used anywhere, no need to add it

File: README.md [hunk 12]
```diff
@@ -51,10 +42,11 @@ See [the list of releases][6] to find out about feature changes.
...
 [24]: https://groups.google.com/forum/#!forum/projectvelero
 [25]: https://kubernetes.slack.com/messages/velero
 [28]: https://velero.io/docs/install-overview
 [29]: https://velero.io/docs/
 [30]: https://velero.io/docs/troubleshooting
 [99]: https://velero.io/docs/support-matrix
-[100]: /site/docs/master/img/velero.png
+[100]: https://velero.io/docs/master/img/velero.png
```
Comment (skriss) [inline README.md]: Doesn't look like this is used anywhere, no need to add it

File: README.md [hunk 13]
```diff
@@ -51,10 +42,11 @@ See [the list of releases][6] to find out about feature changes.
 [11]: https://kubernetes.io/docs/tasks/tools/install-kubectl/#tabset-1
 [12]: https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/README.md
 [14]: https://github.com/kubernetes/kubernetes
+[17]: https://velero.io/docs/master/restic.md
 [24]: https://groups.google.com/forum/#!forum/projectvelero
 [25]: https://kubernetes.slack.com/messages/velero
 [28]: https://velero.io/docs/install-overview
 [29]: https://velero.io/docs/
 [30]: https://velero.io/docs/troubleshooting
 [99]: https://velero.io/docs/support-matrix
```
Comment (skriss) [inline README.md]: probably want to drop link [99] too since the page no longer exists

File: site/docs/master/README.md [hunk 14]
```diff
@@ -53,7 +44,7 @@ See [the list of releases][6] to find out about feature changes.
 [11]: https://kubernetes.io/docs/tasks/tools/install-kubectl/#tabset-1
 [12]: https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/README.md
 [14]: https://github.com/kubernetes/kubernetes
-
+[17]: restic.md
```
Comment (skriss) [inline site/docs/master/README.md]: Doesn't look like this is used anywhere, no need to add it

File: site/docs/master/custom-plugins.md [hunk 15]
```diff
@@ -1,4 +1,4 @@
-# Plugins
+# Custom plugins
```
[the remark this answers is not in the document]
Comment (carlisia) [reply site/docs/master/custom-plugins.md]: Hm. Yeah, now that we are going to have all plugins out of tree there's no more native x custom so I'll change it back.

File: README.md (hunk 12 above)
Comment (carlisia) [reply README.md]: It's on line 1.

Comment (skriss) [reply README.md]: Ah. I don't think it needed to change, but I guess it's fine to hit the website.
